Core/src/org/sleuthkit/autopsy/report/ReportGenerator.java

```diff
@@ -136,16 +136,9 @@ public void windowClosing(WindowEvent e) {
     /**
      * Run the GeneralReportModules using a SwingWorker.
      */
-    void generateGeneralReport(GeneralReportModule generalReportModule) {
+    void generateGeneralReport(GeneralReportModule generalReportModule) throws IOException {
         if (generalReportModule != null) {
-            reportPathFormatString = String.format(reportPathFormatString, generalReportModule.getName());
-            // Create the root reports directory.
-            try {
-                FileUtil.createFolder(new File(reportPathFormatString));
-            } catch (IOException ex) {
-                errorList.add(NbBundle.getMessage(this.getClass(), "ReportGenerator.errList.failedMakeRptFolder"));
-                logger.log(Level.SEVERE, "Failed to make report folder, may be unable to generate reports.", ex); //NON-NLS
-            }
+            reportPathFormatString = createReportDirectory(generalReportModule, reportPathFormatString);
```

Comment:
It would be better to not overwrite the format string with a formatted path. This assumes that the format string will only ever be used once. It would be better to make reportPathFormatString into a static final String REPORT_PATH_FMT_STR and use a local variable for the return value of createReportDirectory.